stored_requests/backends/db_fetcher/fetcher.go

```diff
@@ -4,30 +4,34 @@ import (
 	"context"
 	"database/sql"
 	"encoding/json"
-
 	"github.com/lib/pq"
 
 	"github.com/golang/glog"
 	"github.com/prebid/prebid-server/stored_requests"
 )
 
-func NewFetcher(db *sql.DB, queryMaker func(int, int) string) stored_requests.AllFetcher {
+func NewFetcher(db *sql.DB, queryMaker func(int, int) string, responseQueryMaker func(int) string) stored_requests.AllFetcher {
 	if db == nil {
 		glog.Fatalf("The Postgres Stored Request Fetcher requires a database connection. Please report this as a bug.")
 	}
 	if queryMaker == nil {
 		glog.Fatalf("The Postgres Stored Request Fetcher requires a queryMaker function. Please report this as a bug.")
 	}
+	if responseQueryMaker == nil {
+		glog.Fatalf("The Postgres Stored Response Fetcher requires a responseQueryMaker function. Please report this as a bug.")
+	}
```

Comment:
This code is not clear enough and needs to be separate in future.
Now it works with any kind of configuration because function is hardcoded and logic in upstream function determines what kind of data we want to fetch:
/stored_requests/config/config.go -> func newFetcher :
```
//stored imps, req, videos, responses. If config exists each of them will have a separate fetcher. 
if cfg.Postgres.FetcherQueries.QueryTemplate != "" { 

      idList = append(idList, db_fetcher.NewFetcher(db, cfg.Postgres.FetcherQueries.MakeQuery, cfg.Postgres.FetcherQueries.MakeQueryResponses))
       
}
```
If stored response config is not defined - this part of code will be skipped. 
Hope this answers your question.
But Idk why do we need to check if `queryMaker` and `responseQueryMaker` are defined, they are hardcoded upstream. I followed existing code pattern.

Reply:
Ah gotcha. I see that they are always not nil.